dirsrvtests/tests/suites/clu/dsidm_account_test.py

```diff
@@ -120,8 +135,376 @@ def test_dsidm_account_entry_status_with_lock(topology_st, create_test_user):
+    :steps:
+        1. Run dsidm account rename option on existing account
+        2. Check the account does not have another uid attribute with the old rdn
+        3. Check the old account is deleted
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+    """
+
+    standalone = topology_st.standalone
+    accounts = nsUserAccounts(standalone, DEFAULT_SUFFIX)
+    test_account = accounts.get(test_user_name)
+
+    args = FakeArgs()
+    args.dn = test_account.dn
+    args.new_name = 'renamed_account'
+    args.new_dn = 'uid=renamed_account,ou=people,{}'.format(DEFAULT_SUFFIX)
+    args.keep_old_rdn = False
+
+    log.info('Test dsidm account rename-by-dn')
+    rename(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    new_account = accounts.get(args.new_name)
+
+    try:
+        output = 'Successfully renamed to {}'.format(new_account.dn)
+        check_value_in_log_and_reset(topology_st, check_value=output)
+
+        log.info('Verify the new account does not have a uid attribute with the old rdn')
+        assert not new_account.present('uid', test_user_name)
+        assert new_account.present('displayName', test_user_name)
+
+        log.info('Verify the old account does not exist')
+        assert not test_account.exists()
+    finally:
+        log.info('Clean up')
+        new_account.delete()
+
+
+@pytest.mark.skipif(ds_is_older("1.4.2"), reason="Not implemented")
+def test_dsidm_account_rename_by_dn_keep_old_rdn(topology_st, create_test_user):
+    """ Test dsidm account rename-by-dn option with keep-old-rdn
+
+    :id: a128bdbb-c0a4-4d9d-9a95-9be2d3780094
+    :setup: Standalone instance
+    :steps:
+        1. Run dsidm account rename option on existing account
+        2. Check the account has another uid attribute with the old rdn
+        3. Check the old account is deleted
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+    """
+
+    standalone = topology_st.standalone
+    accounts = nsUserAccounts(standalone, DEFAULT_SUFFIX)
+    test_account = accounts.get(test_user_name)
+
+    args = FakeArgs()
+    args.dn = test_account.dn
+    args.new_name = 'renamed_account'
+    args.new_dn = 'uid=renamed_account,ou=people,{}'.format(DEFAULT_SUFFIX)
+    args.keep_old_rdn = True
+
+    log.info('Test dsidm account rename-by-dn')
+    rename(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    new_account = accounts.get(args.new_name)
+    output = 'Successfully renamed to {}'.format(new_account.dn)
+    check_value_in_log_and_reset(topology_st, check_value=output)
+
+    log.info('Verify the new account does not have a uid attribute with the old rdn')
+    assert new_account.present('uid', test_user_name)
+    assert new_account.present('displayName', test_user_name)
+
+    log.info('Verify the old account does not exist')
+    assert not test_account.exists()
+
+    log.info('Clean up')
+    new_account.delete()
```

Comment:
We might need another `try-finally` here too

Reply:
Oh, I missed that, thanks for pointing it out. Updated the PR with fix.